src/app/Services/SearchService.php

```diff
@@ -12,10 +12,21 @@ class SearchService
      *
      * @return array<string, mixed>
      */
-    public function search(string $query): array
+    public function search(string $query)
     {
         $userResults = User::where('name', 'like', "%$query%")->get();
-        $postResults = Post::where('content', 'like', "%$query%")->get();
+        $searchWords = explode(' ', $query);
+        $postResults = [];
+
+        foreach ($searchWords as $word) {
+            $posts = Post::where('content', 'like', "% $word %")
+                         ->orWhere('content', 'like', "$word %")
+                         ->orWhere('content', 'like', "% $word")
+                         ->orWhere('content', 'like', $word)
+                         ->get();
+
+            $postResults = array_merge($postResults, $posts->all());
+        }
 
```

Comment:
What is that HAHAHAH

Reply:
use whereIn instead

Reply:
Resolved this comment sir. I didn't think that the initial migrate of the contents is in string thats why i cant use REGEX. Thank you sir!

Reply:
Based on your previous code, it calls posts every word and that's not good in terms of performance.